src/rgw/rgw_sal_motr.cc

```diff
@@ -3737,7 +3737,7 @@ int MotrStore::next_query_by_name(string idx_name,
 
 int MotrStore::delete_motr_idx_by_name(string iname)
 {
-  struct m0_idx idx;
+  struct m0_idx idx = {};
```

Comment:
Strange, `m0_idx_init()` below at line 3747 must initialise it.

Reply:
@madhavemuri any idea why `m0_idx_init()` doesn't initialise `m0_idx::in_attr`?

Reply:
@andriytk  m0_idx_init would call m0_entity_init. entity flags needs to be set my caller, during entity init, all entity required parameter would be set but there is no reset/initialization  of flag details because that has to come from motr application.

Reply:
Where is it documented which flags must be set by motr application before `m0_idx_init()` is called?

Reply:
One document for the M0_ENF_META flag is available , this is required to avoid dix lookup

https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/297862297/md-path+optimization+by+avoiding+internal+dix+layout+lookups

For IO,
https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/316310457/io-path+optimization+by+avoiding+a+internal+lookup+mdcob+lookup+for+pool+version

This are the two documents, I am aware of.
@madhavemuri  another document do we have it.

Reply:
I mean, where exactly is it documented in Motr API doc at motr/client.h ?

Reply:
We may need to find a place holder to collect all this details in one location.